javascript-calculator/src/reducers/calculateReducer.js

```diff
@@ -0,0 +1,115 @@
+        case OPERATOR_INPUT:
+            if (prevState.equation.includes("=")) {
+                let value = prevState.display;
+                value += action.payload;
+                equation = value
+            } else if (prevState.equation !== "" && prevState.equation.match(/[ *\- +/]$/) === null) {
+                let value = prevState.equation;
+                value += action.payload;
+                equation = value
+            } else if (prevState.equation.match(/[+\- */]$/) !== null) {
+                if (prevState.equation.includes("+-") === true && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 2)
+                    value += action.payload;
+                    equation = value
+                }
+                else if (prevState.equation.includes("/-") === true && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 2)
+                    value += action.payload
+                    equation = value
+                } else if (prevState.equation.includes("*-") === true && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 2)
+                    value += action.payload
+                    equation = value
+                }
+                else if (prevState.equation.match(/[+/ *]$/) !== null && action.payload === "-") {
+                    let value = prevState.equation + action.payload
+                    equation = value
+                } else if (prevState.equation.match(/[+]/) !== null && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 1)
+                    value += action.payload
+                    equation = value
+                } else if (prevState.equation.match(/-/) !== null && action.payload !== "-") {
+                    let value = prevState.equation;
+                    value = value.substring(0, value.length - 1)
+                    value += action.payload
+                    equation = value
+                }
+            }
+            return { prevState, display, equation }
+        case DECIMAL_INPUT:
+            if (prevState.equation === "" || prevState.equation.includes("=")) {
+                let value = "0."
+                equation = value
+                display = value
+            } else if (prevState.equation.match(/[+\- */]$/)) {
+                let value = "0."
+                display = value
+                equation = prevState.equation + value
+            } else if (!prevState.display.includes(".")) {
+                equation = prevState.equation + action.payload
+                display = prevState.display + action.payload
+            }
+            return { prevState, display, equation }
```

Comment:
Use object spread operator proposal to write `return { ...prevState, display, equation }`